example/src/main/tut/typeclass/Monad.md

```diff
@@ -0,0 +1,49 @@
+---
+layout: docs
+title:  "Monad"
+---
+
+# Monad
+
+*A monad is the combination of an [Applicative](./Applicative.html) with a [Bind](./Bind.html)*
+
+A monad instance needs to satifsy the monad laws:
+
+- Left identity: `pure(a).flatMap(f) === f(a)`
+- Right identity: `x.flatMap(pure) === x`
+- Associativity: `x.flatMap(f).flatMap(g) === m.flatMap(x => f(x).flatMap(g))`
+
+**Typical imports**
+
+```tut:silent
+import scalaz._
+import scalaz.Prelude._
+```
+
+## Instance declaration
+
+```tut
+import scalaz.typeclass.MonadClass
+
+val listMonad: Monad[List] = new MonadClass[List] {
+  def pure[A](a: A): List[A] = List(a)
+  def ap[A, B](fa: List[A])(f: List[A => B]): List[B] = fa.zip(f).map(t => t._2(t._1))
+  def flatMap[A, B](ma: List[A])(f: A => List[B]): List[B] = ma.flatMap(f)
+  def flatten[A](ma: List[List[A]]): List[A] = ma.flatten
+  def map[A, B](ma: List[A])(f: A => B): List[B] = ma.map(f)
+
+}
+```
+
+## Usage
+
+```tut
+val l = List(1, 2, 3).flatMap(x => List(x + 5))
+val k = List(2, 3, 4)
+
+for {
+  x <- l
+  y <- k
+} yield x * y
+```
```

Comment:
👍 for the `for` syntax example!

Reply:
although... isn't this just using List#flatMap? You should wrap this in a def example[F[_]: Monad] or similar.

Reply:
Yeah, the reason I am using `List` is that I can later just replace all the types with `IList` and the examples should still work.

Reply:
Well, but... since scala knows that the receiver of `flatMap` is a `List`, it'll pick the instance method `flatMap` and not the typeclass syntax one (so this isn't even demoing anything scalaz!)

Reply:
I know :)
But it is way easier to use `List` for the moment and replace it with `IList` later than use one of our data types right now. Most of them are still missing good chunks of syntax or type class instances.